### PR TITLE
Add BOM handling to the ingest apps that read CSV files.

### PR DESCRIPTION
**Pull request by malaffoon**
Thanks to apache commons-io this is a simple fix. Most of the code is unit tests and minor cleanup of dependencies management.

### CODE REVIEW

File: group-processor/build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ apply plugin: 'org.springframework.boot'
```
Comment (malaffoon) [inline group-processor/build.gradle]: Mostly unrelated cleanup to make quotes consistent, alphabetize some of the dependencies, and use a common version for the apache commons libs.

**Comment by jtreuting** [comment]
I'm guessing this means we don't need to update the CSV template file provided as this takes care of that?

**Comment by malaffoon** [comment]
@jtreuting that's what i was trying to address, yes.
